elementary/monitor/alerts/group_of_alerts.py

```diff
@@ -0,0 +1,337 @@
+                        tags.update([x.strip() for x in al.tags.split(",")])
+                    elif isinstance(tags_unjsoned, list):  # tags was a list of strings
+                        tags.update(tags_unjsoned)
+                elif isinstance(al.tags, list):
+                    tags.update(al.tags)
+        TAG_PREFIX = "#"
+        formatted_tags = [
+            tag if tag.startswith(TAG_PREFIX) else f"{TAG_PREFIX}{tag}" for tag in tags
+        ]
+        return ", ".join(formatted_tags)
+
+    def _sort_channel_destination(self, default_channel):
+        raise NotImplementedError
+
+    def _fill_components_to_alerts(self, alerts):
+        errors = []
+        warnings = []
+        failures = []
+        for alert in self.alerts:
+            if isinstance(alert, ModelAlert) or alert.status == "error":
+                errors.append(alert)
+            elif alert.status == "warn":
+                warnings.append(alert)
+            else:
+                failures.append(alert)
+        self._components_to_alerts: Dict[AlertGroupComponent, List[Alert]] = {
+            FailureComponent: failures,
+            WarningComponent: warnings,
+            ErrorComponent: errors,
+        }
+
+    def to_slack(self):
+        title_blocks = []  # title, [banner], number of passed or failed,
+        title_blocks.append(self._title_block())
+        banner_block = self._get_banner_block()
+        if banner_block:
+            title_blocks.append(banner_block)
+        title_blocks.append(self._number_of_failed_block())
+        self._add_title_to_slack_alert(title_blocks=title_blocks)
+
+        # attention required : tags, owners, subscribers
+        self._add_preview_to_slack_alert(
+            preview_blocks=self._attention_required_blocks()
+        )
+
+        details_blocks = []
+        for component, alerts_list in self._components_to_alerts.items():
+            details_blocks.append(
+                self.create_text_section_block(
+                    f":{component.emoji_in_summary}: *{component.name_in_summary}*"
+                )
+            )
+            details_blocks.append(self.create_divider_block())
+            if len(alerts_list) == 0:
+                text = f"_{component.empty_section_content}_"
+            else:
+                text = self._tabulate_list_of_alerts(alerts_list)
+            details_blocks.append(self.create_text_section_block(text))
+        self._add_blocks_as_attachments(details_blocks)
+
+        return self.get_slack_message()
+
+    def _title_block(self):
+        return self.create_header_block(
+            f":small_red_triangle: {self._title} ({len(self.alerts)} alerts)"
+        )
+
+    def _number_of_failed_block(self):
+        # small_red_triangle: Falied: 36    |    :Warning: Warning: 3    |    :exclamation: Errors: 1
+        fields = []
+        all_components = list(self._components_to_alerts.items())
```

Comment:
_components_to_alerts is a dict which means that it can have a different order each time.
Because you have only 3 types of fields, I think that adding each one explicitly at the right order is doable and will make sure we have a consistency in our code 🙂

Reply:
maybe OrderedDict